### PR TITLE
fix(exporter): correct quorum detection for committee duties

### PR DESCRIPTION
**Pull request by olegshmuelov**
Fixes quorum detection issues introduced in #2558.

  **Problems Fixed:**

  1. **False positives:** Signatures were checked against both roles, causing attestation signatures to incorrectly trigger sync committee quorum
  2. **Missing detection:** When signatures arrived before proposal, quorum was never checked after flushing pending signatures

  **Solution:**

  - Use role-specific root matching in `checkAndPublishQuorum()` to prevent cross-role false positives
  - Add `checkQuorumAfterFlush()` to detect quorum immediately after `flushPending()` reclassifies buffered signatures
  - Refactor for better code quality and debuggability
  - Comprehensive test coverage for all edge cases

### CODE REVIEW

File: operator/dutytracer/collector.go [hunk 1]
```diff
@@ -616,8 +618,27 @@ func (c *Collector) collect(ctx context.Context, msg *queue.SSVMessage, verifySi
...
+								pendingCount += len(idxs)
+							}
+						}
+					}
+					c.logger.Error("CRITICAL: failed to compute role roots from proposal - pending signatures will be dropped",
+						zap.Error(err),
+						fields.Slot(slot),
+						fields.CommitteeID(committeeID),
+						zap.Int("pending_signature_count", pendingCount),
+						zap.Int("pending_roots_count", pendingRoots))
```
Comment (iurii-ssv) [inline operator/dutytracer/collector.go]: If this is indeed critical, maybe we should log the roots and signatures (and whatever other data we might need) for further debugging ?

We also have the `Warn loudly if we are about to drop pending buffered entries ...` below that seems like it needs a "similar CRITICAL treatment" ?

File: operator/dutytracer/eviction.go [hunk 2]
```diff
@@ -68,9 +78,11 @@ func (c *Collector) dumpCommitteeToDBPeriodically(slot phase0.Slot) (totalSaved
 			// We intentionally drop pending entries here; they are not persisted.
 			trace.pendingByRoot = nil
 		}
+
+		// Deep copy while still holding the lock
+		data := trace.DeepCopy()
 		trace.Unlock()
```
Comment (iurii-ssv) [inline operator/dutytracer/eviction.go:80]: The `DeepCopy` method could use a comment about the "need to hold the lock",

and there is also a `trace()` method that maybe could use a re-name into `DeepCopyLocked` or something

File: operator/dutytracer/eviction.go [hunk 3]
```diff
@@ -80,23 +92,25 @@ func (c *Collector) dumpCommitteeToDBPeriodically(slot phase0.Slot) (totalSaved
...
+	// Delete from map FIRST while collecting data to prevent race condition
+	// where new messages arrive between reading and deletion
 	c.validatorTraces.Range(func(idx phase0.ValidatorIndex, slotToTraceMap *hashmap.Map[phase0.Slot, *validatorDutyTrace]) bool {
 		trace, found := slotToTraceMap.Get(slot)
 		if !found {
 			return true
 		}
 
+		// Remove from map immediately to prevent new messages from modifying this trace
+		slotToTraceMap.Delete(slot)
```
Comment (iurii-ssv) [inline operator/dutytracer/eviction.go]: Nit: I'm not sure if this specific comment (and similar ones) is entirely accurate:
- the `trace` object has a dedicated lock that prevents all concurrent modifications ... so it alone should be able to take care of that, no ?
- the `slotToTraceMap.Delete` here does reduce the likelihood of concurrent actor "acquiring the `slotToTraceMap` object" but doesn't solve it entirely (I think only mutex can solve it entirely)

I do like the additional comments on this PR, just want to make sure they don't "create a false sense of security" about this stuff being atomic (it seems to be "best effort kind of thingy")

File: operator/dutytracer/collector.go (hunk 1 above)
Comment (olegshmuelov) [reply operator/dutytracer/collector.go]: Added root logging: Implemented using the existing `pendingDetails()` helper.
Kept eviction without "CRITICAL:" prefix since it's cleanup, not an active failure, but both provide the same detailed debugging data.

File: operator/dutytracer/eviction.go (hunk 3 above)
Comment (olegshmuelov) [reply operator/dutytracer/eviction.go]: Exactly right on both points!

  1. Yes, the trace's lock is what provides concurrency safety - not the map deletion. The map deletion is purely about lifecycle management (preventing new references).

  2. Correct - deletion doesn't guarantee exclusivity. A goroutine could have already acquired a reference via `Get(slot)` before we delete.
  Updated comments in all 3 dump functions to reflect this
  
8b6781d2396b0d32e2d7bc0431dba5464c0b71b0
